urld5c972ef81_noxx90_jflam_tgz-pJ8-fr_noxx90_jflam_BenchmarkJ8/src/fr/noxx90/jflam/FlameLogger.java

```diff
@@ -0,0 +1,63 @@
+package fr.noxx90.jflam;
+
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import com.esotericsoftware.minlog.Log.Logger;
+
+public class FlameLogger extends Logger {
+	private SimpleDateFormat format;
+	private PrintWriter writer;
+	
+	public FlameLogger() {
+		format = new SimpleDateFormat("HH:mm:ss");
+		try {
+			writer = new PrintWriter("log");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *writer*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).